minio/api.py

```diff
@@ -34,46 +33,40 @@
 from urllib.parse import urlunsplit
 from xml.etree import ElementTree as ET
 
-import certifi
-import urllib3
-from urllib3._collections import HTTPHeaderDict
+
+from minio.http import (HttpClient, _DEFAULT_USER_AGENT,
+                        convert_to_urllib3_headers)
 
 from . import __title__, __version__, time
 from .commonconfig import COPY, REPLACE, ComposeSource, CopySource, Tags
-from .credentials import StaticProvider
 from .datatypes import (CompleteMultipartUploadResult, EventIterable,
                         ListAllMyBucketsResult, ListMultipartUploadsResult,
                         ListPartsResult, Object, Part, PostPolicy,
                         parse_copy_object, parse_list_objects)
 from .deleteobjects import DeleteError, DeleteRequest, DeleteResult
 from .error import InvalidResponseError, S3Error, ServerError
 from .helpers import (MAX_MULTIPART_COUNT, MAX_MULTIPART_OBJECT_SIZE,
-                      MAX_PART_SIZE, MIN_PART_SIZE, BaseURL, ObjectWriteResult,
+                      MAX_PART_SIZE, MIN_PART_SIZE, ObjectWriteResult,
                       ThreadPool, check_bucket_name, check_non_empty_string,
                       check_sse, check_ssec, genheaders, get_part_info,
                       headers_to_strings, is_valid_policy_type, makedirs,
-                      md5sum_hash, read_part_data, sha256_hash)
+                      md5sum_hash, read_part_data)
 from .legalhold import LegalHold
 from .lifecycleconfig import LifecycleConfig
 from .notificationconfig import NotificationConfig
 from .objectlockconfig import ObjectLockConfig
 from .replicationconfig import ReplicationConfig
 from .retention import Retention
 from .select import SelectObjectReader, SelectRequest
-from .signer import presign_v4, sign_v4_s3
+from .signer import presign_v4
 from .sse import SseCustomerKey
 from .sseconfig import SSEConfig
 from .tagging import Tagging
 from .versioningconfig import VersioningConfig
 from .xml import Element, SubElement, findtext, getbytes, marshal, unmarshal
 
-_DEFAULT_USER_AGENT = (
-    f"MinIO ({platform.system()}; {platform.machine()}) "
-    f"{__title__}/{__version__}"
-)
-
 
-class Minio:  # pylint: disable=too-many-public-methods
+class Minio(HttpClient):  # pylint: disable=too-many-public-methods
```

Comment:
Nothing needs to be changed in this class. You would need to have separate methods inside `MinioAdmin` class.

Reply:
Ok, but you suggested to make a base class before